src/modules/townhalls/townhalls.ts

```diff
@@ -66,8 +72,130 @@ export async function getBillInfo(townhallId: string) {
+        promises4 = senateUrls.map((url) => getVoteResult(url));
+        senateResponse = await Promise.all(promises4);
+    }
+
+    for (let i = 0; i < billResponses.length; i += 1) {
+        for (let j = 0; j < houseResponse.length; j += 1) {
+            if (
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                houseResponse[j].data.results.votes.vote.bill.bill_id ===
+                billResponses[i].billId
+            ) {
+              
+                for (
+                    let y = 0;
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                    y < houseResponse[j].data.results.votes.vote.positions.length;
+                    y += 1
+                ) {
+                    if (
+                        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                        houseResponse[j].data.results.votes.vote.positions[y].name ===
+                        townhall?.form.speaker
+                    ) {
+                        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                        billResponses[i].vote_position =
+                            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                            houseResponse[j].data.results.votes.vote.positions[
+                                y
+                            ].vote_position;
+                    }
+                }
+            }
+        }
+    }
+    for (let i = 0; i < billResponses.length; i += 1) {
+        for (let j = 0; j < senateResponse.length; j += 1) {
+            if (
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                senateResponse[j].data.results.votes.vote.bill.bill_id ===
+                billResponses[i].billId
+            ) {
+              
+                for (
+                    let y = 0;
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                    y < senateResponse[j].data.results.votes.vote.positions.length;
+                    y += 1
+                ) {
+                    if (
+                        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                        senateResponse[j].data.results.votes.vote.positions[y].name ===
+                        townhall?.form.speaker
+                    ) {
+                        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                        billResponses[i].vote_position =
+                            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                            senateResponse[j].data.results.votes.vote.positions[
+                                y
+                            ].vote_position;
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Could be abstracted to a helper function.  The entirety of this code could also be abstracted to a helper function.  Ex. `const billInfoHouse = extractBillInfo(housevariableshere);`
`const billInfoSenate = extractBillInfo(senatevariableshere);`